pkg/validate/validate.go

```diff
@@ -32,19 +33,108 @@ func Validate(mb utils.MetaBundle) (bool, []error) {
 }
 
-func GetAllValidators() []utils.Validator {
-	return []utils.Validator{
-		{
-			Description: "Ensure defaultChannel is present in list of channels",
-			Runner:      validators.ValidateDefaultChannel,
-		},
-		{
-			Description: "Ensure `label` to follow the format api.openshift.com/addon-<operator-id>",
-			Runner:      validators.ValidateAddonLabel,
-		},
-		{
-			Description: "Ensure current csv is present in the index image",
-			Runner:      validators.ValidateCSVPresent,
-		},
+// name formatting rule: [0-9]{3}_([a-z]*_?)*
+var AllValidators = map[string]utils.Validator{
+	"001_default_channel": {
+		Description: "Ensure defaultChannel is present in list of channels",
+		Runner:      validators.ValidateDefaultChannel,
+	},
+	"002_label_format": {
+		Description: "Ensure `label` to follow the format api.openshift.com/addon-<operator-id>",
+		Runner:      validators.ValidateAddonLabel,
+	},
+	"003_csv_present": {
+		Description: "Ensure current csv is present in the index image",
+		Runner:      validators.ValidateCSVPresent,
+	},
+}
+
+func GetExistingValidatorNames() string {
+	var res []string
+	for name := range AllValidators {
+		res = append(res, name)
+	}
+	return strings.Join(res, ",")
+}
+
+type validatorsFilter struct {
+	ReturnAll      bool
+	Enabled        bool
+	ValidatorNames []string
+}
+
+func NewFilter(disabled, enabled string) (*validatorsFilter, error) {
+	// error: mutually exclusive
+	if disabled != "" && enabled != "" {
+		return nil, errors.New("Can't set both --disabled and --enabled. They are mutually exclusive.")
+	}
+	// empty filter
+	if disabled == "" && enabled == "" {
+		return &validatorsFilter{
+			ReturnAll: true,
+		}, nil
+	}
+	var validatorNames []string
+	if enabled != "" {
+		validatorNames = strings.Split(enabled, ",")
+	} else {
+		validatorNames = strings.Split(disabled, ",")
+	}
+
+	validNames := GetExistingValidatorNames()
+	for _, name := range validatorNames {
+		if _, ok := AllValidators[name]; !ok {
+			return nil, fmt.Errorf("Could not find validator with name %v. Existing validator are %v.", name, validNames)
+		}
+	}
+
+	return &validatorsFilter{
+		Enabled:        enabled != "",
+		ValidatorNames: validatorNames,
+	}, nil
+}
+
+func (f *validatorsFilter) GetValidators() []utils.Validator {
+	if f.ReturnAll {
+		return f.getAllValidators()
+	}
+	if f.Enabled {
+		return f.getEnabledValidators()
+	}
+	return f.getDisabledValidators()
+}
+
+func (f *validatorsFilter) getAllValidators() []utils.Validator {
+	var res []utils.Validator
+	for _, validator := range AllValidators {
+		res = append(res, validator)
+	}
+	return res
+}
+
+func (f *validatorsFilter) getEnabledValidators() []utils.Validator {
+	var res []utils.Validator
+	for _, name := range f.ValidatorNames {
+		// can ignore _ok_ was already checked
+		validator, _ := AllValidators[name]
+		res = append(res, validator)
+	}
+	return res
+}
+
+func (f *validatorsFilter) getDisabledValidators() []utils.Validator {
```

Comment:
```suggestion
func (f *validatorsFilter) getNotDisabledValidators() []utils.Validator {
```
Or maybe we can change it some other more suitable name? `getDisabledValidators` sounds like something which would return validators which are disabled but in reality, the method is doing the opposite.
wdyt :)

Reply:
refactored, this function is gone